pkg/vmprovider/providers/vsphere2/vcenter/getvm.go

```diff
@@ -31,14 +31,11 @@ func GetVirtualMachine(
 		}
 	}
 
-	// For when we start to use the k8s VM.UID for the VC VM's InstanceUUID or UUID (aka BiosUUID):
-	/*
-		if instanceUUID := vmCtx.VM.UID; instanceUUID != "" {
-			if vm, err := findVMByUUID(vmCtx, vimClient, datacenter, string(instanceUUID), true); err == nil {
-				return vm, nil
-			}
+	if instanceUUID := vmCtx.VM.UID; instanceUUID != "" {
```

Comment:
Do we want to do this lookup first since it is support to be the most stable?

Should we only do this lookup when this UID matches the Spec value?

Reply:
Yes, changed the order.

For new VMs:
- `InstanceUuid` is set to the VM's `ObjectMeta.UID`
- `Uuid` is set to VM's spec.biosUUID

For existing brownfield VMs, both InstanceUuid and Uuid were generated by VC. So I don't think we have a way to skip? But, we could add lookup for BiosUUID, as in:
```
findVMByUUID(vmCtx, vimClient, datacenter, vmCtx.VM.Spec.BiosUUID, false)
```
What do you think?